legend-engine-language-pure-compiler/src/main/java/org/finos/legend/engine/language/pure/compiler/toPureGraph/ProcessingContext.java

```diff
@@ -26,8 +26,25 @@
 public class ProcessingContext
 {
     public MutableList<MutableMap<String, ValueSpecification>> inferredVariableList = FastList.newList();
-    public MilestoningDatePropagationContext milestoningDatePropagationContext = new MilestoningDatePropagationContext();
+    public Stack<MilestoningDatePropagationContext> milestoningDatePropagationContext = new Stack<>();
     private final Stack<String> tags = new Stack<>();
+    public boolean isDatePropagationSupported = true;
+
+    public void pushMilestoningDatePropagationContext(MilestoningDatePropagationContext milestoningContext)
+    {
+        MilestoningDatePropagationContext context = new MilestoningDatePropagationContext(milestoningContext.getProcessingDate(), milestoningContext.getBusinessDate());
```

Comment:
Why creating a new context?  Cannot we make MilestoningDatePropagationContext immutable?  It seems that you remove the need of the setters on with this change.  

Note that you are anyways exposing the stack references thru the peek, hence ensuring state cannot change might be required.